File: CodeForces/1101-1200/1154/A/1154A.py
# Problem 1154A: Restoring Three Numbers
# https://codeforces.com/contest/1154/problem/A
# Executed with: Python 3.11.2
# Linux version: Linux kernel 6.1.0-10-amd64
# Submitted on: September 27th, 2024
# Codeforces language used: PyPy 3.10 (7.3.15, 64bit)

board = sorted(list(map(int, input().split())))

print(f"{board[3]-board[2]} {board[3]-board[1]} {board[3]-board[0]}")

File: CodeForces/1101-1200/1154/A/1154A.ts
// Problem 1154A: Restoring Three Numbers
// https://codeforces.com/contest/1154/problem/A
// Compiled with: npx 9.2.0, tsc Version 5.9.3
// Executed with: node v20.19.4
// Linux version: Linux kernel 6.17.0-5-generic
// Written on: October 26th, 2025
// Codeforces language used: Cannot be submitted

import * as readline from 'node:readline/promises';
import { stdin as input, stdout as output } from 'node:process';

async function main() {
    const rl = readline.createInterface({ input, output });

    const all_nums: string = await rl.question('');
    const pass_info: number[] = all_nums
        .trim()
        .split(' ')
        .map(s => parseInt(s, 10));
    rl.close();

    var sortedArray: number[] = pass_info.sort((n1,n2) => n1 - n2);
    var ans = "";
    ans += sortedArray[3] - sortedArray[2];
    ans += " ";
    ans += sortedArray[3] - sortedArray[1];
    ans += " ";
    ans += sortedArray[3] - sortedArray[0];

    console.log(ans);
}

main();
